Samples/PowerShell/functions/func_ModernDashboards.ps1

```diff
@@ -4,6 +4,134 @@
+
+    This command list all of the Modern Dashboards
+.EXAMPLE
+    $SwisConnection = Connect-Swis -Hostname "192.168.11.165" -Username "admin" -Password "MyComplexPassword"
+    PS C:\> Get-ModernDashboard -SwisConnection $SwisConnection -DashboardId 9 
+
+    Lists Modern Dashboard with ID 9
+.EXAMPLE
+    $SwisConnection = Connect-Swis -Hostname "192.168.11.165" -Username "admin" -Password "MyComplexPassword"
+    PS C:\> Get-ModernDashboard -SwisConnection $SwisConnection -DashboardId 1 
+
+    PS C:\> Get-ModernDashboard -SwisConnection $SwisConnection -DashboardId 1 -IncludeSystem
+
+    The first command will (most likely) return nothing as DashboardId 1 is a "system" dashboard
+    The second command will return the details of Dashboard ID 1
+
+.NOTES
+    Author:  Kevin M. Sparenberg (https://thwack.solarwinds.com/members/kmsigma)
+    Version: 0.9
+    Last Updated: 2021-09-30
+    Validated: Orion Platform 2020.2.5
+
+    TBD List:
+        * Validate that the script works on non-Windows
+#>
+function Get-SwisModernDashboard {
+    [CmdletBinding(
+        DefaultParameterSetName = 'Normal', 
+        PositionalBinding = $false,
+        HelpUri = 'https://documentation.solarwinds.com/en/success_center/orionplatform/content/core-fusion-dashboard-import-export.htm',
+        ConfirmImpact = 'Low')]
+    [Alias('Get-ModernDashboard')]
+    [OutputType()]
+    Param
+    (
+        # The connection to the SolarWinds Information Service
+        [Parameter(
+            Mandatory = $false, 
+            ValueFromPipeline = $true,
+            ValueFromPipelineByPropertyName = $true, 
+            ValueFromRemainingArguments = $false)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [Alias("Swis")] 
+        [SolarWinds.InformationService.Contract2.InfoServiceProxy]$SwisConnection = $Global:SwisConnection,
+
+        # The dashboard Id we'll display
+        [AllowNull()]
+        [Alias("Id")] 
+        [int32[]]$DashboardId,
+
+        # Filter for specific owners?
+        [AllowNull()]
+        [string[]]$Owner,
+
+        # Should we include system Dashboards?
+        [AllowNull()]
+        [switch]$IncludeSystem,
+
+        # Omits white space and indented formatting in the output string.
+        [AllowNull()]
+        [switch]$IncludeJson
+
+    )
+
+    Begin {
+        # Define the Property List
+        $PropertyList = "DashboardID", "DisplayName", "UniqueKey", "Owner", "LastUpdate", "Private"
+
+        # Define default Query
+        $Swql = "SELECT DashboardID, ParentID, Owner, UniqueKey, LastUpdate, Private, IsSystem, DisplayName FROM Orion.Dashboards.Instances WHERE ParentID IS NULL"
+
+    }
+    Process {
+        
+        #region Filter for DashboardIds
+        # Add Dashboards if we are sent IDs
+        if ( $DashboardId ) {
+            # Add an = to the WHERE clause
+            $Swql += " AND DashboardID = $( $DashboardId )"
```

Comment:
This seems to assume that we'll receive zero or one dashboard IDs.  If we get 2 or more, this would need to be an `IN` clause instead.

Reply:
Nice catch - fixed.